src/main/java/com/jagrosh/jdautilities/commandclient/CommandBuilder.java

```diff
@@ -0,0 +1,453 @@
+    }
+
+    /**
+     * Sets the {@link com.jagrosh.jdautilities.commandclient.Command#children children}
+     * of the Command built from this CommandBuilder.
+     *
+     * @param  children
+     *         The children of the Command to be built.
+     *
+     * @return This CommandBuilder
+     */
+    public CommandBuilder setChildren(Collection<Command> children)
+    {
+        if(children == null)
+            this.children = new Command[0];
+        else
+            this.children = (Command[]) children.toArray();
+        return this;
+    }
+
+    /**
+     * Sets the {@link com.jagrosh.jdautilities.commandclient.Command#helpBiConsumer
+     * help BiConsumer} of the Command built from this CommandBuilder.
+     *
+     * @param  helpBiConsumer
+     *         The help BiConsumer of the Command to be built.
+     *
+     * @return This CommandBuilder
+     */
+    public CommandBuilder setHelpBiConsumer(BiConsumer<CommandEvent, Command> helpBiConsumer)
+    {
+        this.helpBiConsumer = helpBiConsumer;
+        return this;
+    }
+
+    /**
+     * Sets the Command built to {@link com.jagrosh.jdautilities.commandclient.Command#usesTopicTags
+     * use TopicTags}.
+     *
+     * @param  usesTopicTags
+     *         {@code true} if the Command built is uses topic tags, {@code false} if it does not.
+     *
+     * @return This CommandBuilder
+     */
+    public CommandBuilder setUsesTopicTags(boolean usesTopicTags)
+    {
+        this.usesTopicTags = usesTopicTags;
+        return this;
+    }
+
+    /**
+     * Sets the {@link com.jagrosh.jdautilities.commandclient.Command#cooldownScope
+     * cooldown scope} of the Command built from this CommandBuilder.
+     *
+     * @param  cooldownScope
+     *         The CooldownScope of the Command to be built.
+     *
+     * @return This CommandBuilder
+     */
+    public CommandBuilder setCooldownScope(CooldownScope cooldownScope)
+    {
+        if(cooldownScope == null)
+            this.cooldownScope = CooldownScope.USER;
+        else
+            this.cooldownScope = cooldownScope;
+        return this;
+    }
+
+    /**
+     * Builds the {@link com.jagrosh.jdautilities.commandclient.Command Command}
+     * using the previously provided information.
+     *
+     * <p>This uses the only the {@link com.jagrosh.jdautilities.commandclient.CommandEvent
+     * CommandEvent} parameter that would be provided during
+     * {@link com.jagrosh.jdautilities.commandclient.Command#execute(CommandEvent) #execute(CommandEvent)},
+     * and no information about the Command can be retrieved using this.
+     *
+     * <p>An alternate method {@link #build(BiConsumer)} exists if you wish to retrieve information
+     * about the Command built during execution.
+     *
+     * @param  execution
+     *         The {@link Consumer} that runs on Command#execute(CommandEvent).
+     *
+     * @return The Command built
+     */
+    public Command build(Consumer<CommandEvent> execution)
```

Comment:
Why is the execution consumer an argument of `build()` and not just a setter like all other properties?

Reply:
Because `Command#execute(CommandEvent)` is an abstract method that must be implemented when a class extends `Command`, I thought it would be a stylistic way to mimic this.

The premise is that it is a requirement whether you extend `Command` or use `CommandBuilder` you should have to implement the execution.